app/controllers/api_controller/reports.rb

```diff
@@ -1,12 +1,20 @@
 class ApiController
   module Reports
+    SCHEDULE_ATTR = %w(start_date interval time_zone send_email).freeze
     #
     # Reports Supporting Methods
     #
     def results_query_resource(object)
       object.send("miq_report_results")
     end
 
+    def schedules_query_resource(object)
+      klass = collection_class(:schedules)
+      exp = MiqExpression.new("=" => {"field" => "MiqReport.id",
+                                      "value" => object.id})
+      object ? klass.where(:filter => exp) : {}
```

Comment:
If `object` is falsey here, can you send `id` to it on the line above?

I'm wondering if some of this code could be pushed down into `MiqSchedule`, partly so it could be tested with a little more isolation?